apps/events/forms.py

```diff
@@ -107,6 +107,21 @@ class Meta:
         fields = ["action_type"]
         labels = {"action_type": "Then..."}
 
+    def clean(self):
+        cleaned_data = super().clean()
+        action_type = cleaned_data.get("action_type")
+
+        if self.data and "type" in self.data:
+            trigger_type = self.data.get("type")
+            if trigger_type == "new_bot_message" and action_type in [
+                "send_message_to_bot",
+                "pipeline_start",
+                "summarize",
+                "schedule_trigger",
+            ]:
```

Comment:
Can you say more about why these ones were chosen? I don't think `summarize` can cause a loop, or `pipeline_start` - but I could be wrong. Not sure about `schedule_trigger` either.

Reply:
Pipeline start will cause a loop if it sends a bot message and the trigger is `new_bot_message`. Same with `schedule_trigger`. `summarize` on the other hand should be fine

Reply:
`pipeline_start` can't send messages. The output is just logged: https://github.com/dimagi/open-chat-studio/blob/46a93dfa1fb1b9413caf4497bb8da16318e950c5/apps/events/models.py#L123-L124

Fine to keep `schedule_trigger`.

**Side Note**: `log` and `summarize` can be removed from code since they don't actually do anything useful.